--- gst-plugins-0.8.7/common/m4/gst-feature.m4.orig	2004-09-21 16:16:36.000000000 +0200
+++ gst-plugins-0.8.7/common/m4/gst-feature.m4	2005-01-24 13:28:43.871500536 +0100
@@ -133,7 +133,7 @@
     [$1]_CFLAGS=
     HAVE_[$1]=no
   else
-    if [$2] --plugin-libs [$3] &> /dev/null; then
+    if [$2] --plugin-libs [$3] > /dev/null 2> /dev/null; then
       [$1]_LIBS=`[$2] --plugin-libs [$3]`
     else
       [$1]_LIBS=`[$2] --libs [$3]`
